app/routes.py

```
@@ -369,14 +488,22 @@ def generate_persona_image(request: PersonaImageRequest) -> PersonaImageResponse
         prompt += ". High quality, single person, only face and shoulders, centered, neutral background, avoid borders."
 
         # Generate the image
-        image_response = image_service.generate_persona_image(
+        image_generation_result = image_service.generate_persona_image(
             name=name, prompt=prompt, style="realistic", size=(512, 512)
         )
 
+        if not image_generation_result["success"]:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=f"Failed to generate image: {image_generation_result.get('error', 'Unknown error')}"
```

Pylint (3.3.7)

[convention] Line too long (107/100)

(C0301)